yt/visualization/volume_rendering/scene.py

```diff
@@ -437,15 +443,55 @@ def save_annotated(
         """
         fname = self._setup_save(fname, render)
 
-        # which transfer function?
-        rs = self._get_render_sources()[0]
-        tf = rs.transfer_function
-        label = rs.data_source.ds._get_field_info(rs.field).get_label()
-
         ax = self._show_mpl(
             self._last_render.swapaxes(0, 1), sigma_clip=sigma_clip, dpi=dpi
         )
-        self._annotate(ax.axes, tf, rs, label=label, label_fmt=label_fmt)
+
+        # number of transfer functions?
+        num_trans_func = 0
+        for rs in self._get_render_sources():
+            try:
+                tf = rs.transfer_function
+                num_trans_func += 1
+            except AttributeError:
+                print("error")
+                pass
+
+        # which transfer function?
+        if num_trans_func == 1:
+            rs = self._get_render_sources()[0]
+            tf = rs.transfer_function
+            label = rs.data_source.ds._get_field_info(rs.field).get_label()
+            self._annotate(ax.axes, tf, rs, label=label, label_fmt=label_fmt)
+        else:
+            # set the origin and width and height of the colorbar region
+            if tf_rect is not None:
+                try:
+                    cbx0, cby0, cbw, cbh = tf_rect
+                except ValueError:
+                    print(
+                        "tf_rect does not have correct number of values.  Using default"
+                    )
+                    tf_rect = None
```

Comment:
I don't think it's necessary or helpful to catch this error.
```suggestion
                cbx0, cby0, cbw, cbh = tf_rect
```

Reply:
okay, that's fair.